caravel/assets/javascripts/explorev2/components/AdvancedAnalytics.jsx

```diff
@@ -0,0 +1,135 @@
+              name="select-rolling"
+              options={rollingOptions.map((o) => ({ value: o, label: o }))}
+              value={this.props.rolling}
+              autosize={false}
+              onChange={this.changeSelectData.bind(this, 'rolling')}
+            />
+          </div>
+          <div className="col-sm-6">
+            <h5 className="section-heading">Periods</h5>
+            <input
+              type="text"
+              onChange={this.changeInput.bind(this, 'periods')}
+              className="form-control input-sm"
+            />
+          </div>
+          <h5 className="section-heading">Time Shift</h5>
+          <input
+            type="text"
+            onChange={this.changeInput.bind(this, 'timeShift')}
+            className="form-control input-sm"
+          />
+          <div className="col-sm-6">
+            <h5 className="section-heading">Period Ratio</h5>
+            <input
+              type="text"
+              onChange={this.changeInput.bind(this, 'periodRatio')}
+              className="form-control input-sm"
+            />
+          </div>
+          <div className="col-sm-6">
+            <h5 className="section-heading">Period Ratio Type</h5>
+            <Select
+              name="select-rolling"
+              options={periodRatioTypeOptions.map((o) => ({ value: o, label: o }))}
+              value={this.props.periodRatioType}
+              autosize={false}
+              onChange={this.changeSelectData.bind(this, 'periodRatioType')}
+            />
+          </div>
+          <div className="col-sm-6">
+            <h5 className="section-heading">Resample How</h5>
+            <Select
+              name="select-rolling"
+              options={resampleHowOptions.map((o) => ({ value: o, label: o }))}
+              value={this.props.resampleHow}
+              autosize={false}
+              onChange={this.changeSelectData.bind(this, 'resampleHow')}
+            />
+          </div>
+          <div className="col-sm-6">
+            <h5 className="section-heading">Resample Rule</h5>
+            <Select
+              name="select-rolling"
+              options={resampleRuleOptions.map((o) => ({ value: o, label: o }))}
+              value={this.props.resampleRule}
+              autosize={false}
+              onChange={this.changeSelectData.bind(this, 'resampleRule')}
+            />
+          </div>
+          <h5 className="section-heading">Resample Fill Method</h5>
+          <Select
+            name="select-rolling"
+            options={resampleFillOptions.map((o) => ({ value: o, label: o }))}
+            value={this.props.resampleFill}
+            autosize={false}
+            onChange={this.changeSelectData.bind(this, 'resampleFill')}
+          />
+        </div>
+      </div>
+    );
+  }
+}
+
+AdvancedAnalytics.propTypes = propTypes;
+AdvancedAnalytics.defaultProps = defaultProps;
+
+function mapStateToProps(state) {
+  return {
+    rolling: state.rolling,
```

Comment:
I am not sure if they should be top level args in the redux store, maybe `state.viz.advancedAnalytics` or `state.explore.advancedAnalytics`